src/hashing.rs

```diff
@@ -628,6 +642,26 @@ mod test {
         assert_eq!(MacDomain::domain_separation_tag("test"), "com.tari.mac.v1.test");
     }
 
+    #[test]
+    fn finalize_into() {
+        hash_domain!(TestHasher, "com.example.test");
+        let mut hasher = DomainSeparatedHasher::<Blake256, TestHasher>::new();
+        hasher.update([0, 0, 0]);
+
+        let mut output = GenericArray::<u8, <Blake256 as Digest>::OutputSize>::default();
+        hasher.finalize_into(&mut output);
```

Comment:
Maybe add a new line with an assertion to compare the result of output with the same data, hashed using `finalize`.